datalad/distributed/tests/test_drop.py

```diff
@@ -548,3 +549,21 @@ def test_drop_uninit_annexrepo(origpath, path):
     ds = Dataset(path)
     assert_status('ok', ds.drop(what='datasets'))
     nok_(ds.is_installed())
+
+
+# https://github.com/datalad/datalad/issues/6577
+@with_tempfile
+def test_drop_allkeys_result_contains_annex_error_messages(path):
+    # when calling drop with allkeys, expect git-annex error
```

Comment:
Actually: `what=allkeys` is irrelevant from my POV. This may be, where it currently happens, but the change should generally ensure that we get whatever comes from annex regardless.